src/features/game/events/removeCrop.ts

```diff
@@ -0,0 +1,105 @@
+import Decimal from "decimal.js-light";
+import { screenTracker } from "lib/utils/screen";
+import { CROPS } from "../types/crops";
+import { GameState, InventoryItemName } from "../types/game";
+import { isReadyToHarvest, isShovelStolen } from "./harvest";
+
+export enum REMOVE_CROP_ERRORS {
+  SHOVEL_STOLEN = "Shovel stolen!",
+  NO_SHOVEL_SELECTED = "No shovel selected!",
+  NO_SHOVEL_AVAILABLE = "No shovel available!",
+  FIELD_DOESNT_EXIST = "Field doesn't exist!",
+  LOCKED_LAND = "Goblin land!",
+  NO_CROP_PLANTED = "There is no crop planted!",
+  READY_TO_HARVEST = "Plant is ready to harvest!",
+  INVALID_PLANT = "Invalid Plant!",
+}
+
+export type RemoveCropAction = {
+  type: "item.removed";
+  item?: InventoryItemName;
+  index: number;
+};
+
+type Options = {
+  state: GameState;
+  action: RemoveCropAction;
+  createdAt?: number;
+};
+
+export function removeCrop({ state, action, createdAt = Date.now() }: Options) {
+  const fields = { ...state.fields };
+
+  if (isShovelStolen()) {
+    throw new Error(REMOVE_CROP_ERRORS.SHOVEL_STOLEN);
+  }
+
+  if (action.item !== "Rusty Shovel") {
+    throw new Error(REMOVE_CROP_ERRORS.NO_SHOVEL_SELECTED);
+  }
+
+  const shovelAmount = state.inventory["Rusty Shovel"] || new Decimal(0);
+  if (shovelAmount.lessThan(1)) {
+    throw new Error(REMOVE_CROP_ERRORS.NO_SHOVEL_AVAILABLE);
+  }
+
+  if (action.index < 0) {
+    throw new Error(REMOVE_CROP_ERRORS.FIELD_DOESNT_EXIST);
+  }
+
+  if (!Number.isInteger(action.index)) {
+    throw new Error(REMOVE_CROP_ERRORS.FIELD_DOESNT_EXIST);
+  }
+
+  if (
```

Comment:
Do we need to check the locked fields since there won't be any growing crops on them?

All of the crops are already to harvest on the locked land, which means the clauses below should throw and error.

Reply:
@0xsunflowerstudios you are correct - I overlooked this detail. Will remove this logic and irrelevant tests next push :+1: